server/routes/api/landing_page.py

```diff
@@ -518,21 +545,63 @@ def populate_category_data(category):
             # If there is no data for a category in overview page, need to
             # "borrow" it from the category page.
             if not has_data(data):
-                populate_category_data(category)
-                total_charts = 0
-                spec_and_stat[OVERVIEW][category] = []
-                for topic in spec_and_stat[category]:
-                    spec_and_stat[OVERVIEW][category].extend(
-                        spec_and_stat[category][topic])
-                    total_charts += len(spec_and_stat[category][topic])
-                    if total_charts > MAX_OVERVIEW_CHART_GROUP:
-                        break
+                if new_landing_page_cache_flag == "false":
+                    populate_category_data(category, all_stat, spec_and_stat)
+                    total_charts = 0
+                    spec_and_stat[OVERVIEW][category] = []
+                    for topic in spec_and_stat[category]:
+                        spec_and_stat[OVERVIEW][category].extend(
+                            spec_and_stat[category][topic])
+                        total_charts += len(spec_and_stat[category][topic])
+                        if total_charts > MAX_OVERVIEW_CHART_GROUP:
+                            break
+                else:
+                    # if using the new cache, need to make another cache call
+                    # to get data to be borrowed from category page
```

Comment:
For my understanding, what does data to be borrowed mean?  Elaborate?

Reply:
Can we also please pull this out into a helper for testability.

Reply:
There are cases where we have data for a category but none of those stat vars are configured to be shown on overview page (i.e. isOverview = true). For such cases, we borrow a few charts from the category page to show on the overview page.